### PR TITLE
Add Peer Logger

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Cosmetic Addition

**What does this PR do? Why is it needed?**

 - [x] Adds in a peer logger so that the current peer count is logged out along with any relevant data.
 
**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

### CODE REVIEW

**Review by farazdagi — APPROVED** [review]

File: beacon-chain/p2p/service.go [hunk 1]
```diff
@@ -229,6 +232,13 @@ func (s *Service) Start() {
 	runutil.RunEvery(s.ctx, refreshRate, func() {
 		s.RefreshENR()
 	})
+	runutil.RunEvery(s.ctx, 1*time.Minute, func() {
```
Comment (terencechain) [inline beacon-chain/p2p/service.go:235]: Will 1 minute be too verbose?

File: beacon-chain/p2p/peers/status.go [hunk 2]
```diff
@@ -369,6 +369,32 @@ func (p *Status) Connected() []peer.ID {
 	return peers
 }
 
+// Inbound returns the current batch of inbound peers that are connected.
+func (p *Status) Inbound() []peer.ID {
+	p.store.RLock()
```
Comment (terencechain) [inline beacon-chain/p2p/peers/status.go:374]: Would it be better to make the critical section shorter? 
```go
p.store.RLock()
peers := p.store.Peers() 
p.store.RUnlock()
for pid, peerData := range peers { }
```

File: beacon-chain/p2p/service.go (hunk 1 above)
Comment (nisdas) [reply beacon-chain/p2p/service.go:235]: With info logs, you will see it logged out every 5 blocks which is fine imo. Do you think it should be higher ? I can bump it up to 2 minutes in that case.

File: beacon-chain/p2p/peers/status.go (hunk 2 above)
Comment (nisdas) [reply beacon-chain/p2p/peers/status.go:374]: The locking follows the general pattern for all the peer handler methods in the package. Given that this is a read only lock, it shouldn't be too bad having it locked this way, if you feel strongly I can change it to the above pattern.

**Review by terencechain — APPROVED** [review]

File: beacon-chain/p2p/service.go (hunk 1 above)
Comment (terencechain) [reply beacon-chain/p2p/service.go:235]: Sounds good, we can always adjust it later

File: beacon-chain/p2p/peers/status.go (hunk 2 above)
Comment (terencechain) [reply beacon-chain/p2p/peers/status.go:374]: No preference from my end. It was just a thought. Sounds good to me then

**Pull request merged**